addons/mail/static/src/core/thread_actions.js

```diff
@@ -0,0 +1,111 @@
+/* @odoo-module */
+
+import { useComponent, useState } from "@odoo/owl";
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+
+export const threadActionsRegistry = registry.category("mail.thread/actions");
+
+threadActionsRegistry.add("close", {
+    condition(component) {
+        return component.props.chatWindow;
+    },
+    icon: "fa fa-fw fa-close",
+    name: _t("Close Chat Window"),
+    open(component) {
+        component.close();
+    },
+    sequence: 100,
+});
+
+function transformAction(component, id, action) {
+    return {
+        /** Closes this action. */
+        close() {
+            if (this.toggle) {
+                component.threadActions.activeAction = null;
+            }
+            action.close?.(component, this);
+        },
+        /** Optional component that should be displayed in the view when this action is active. */
+        component: action.component,
+        /** Condition to display the component of this action. */
+        get componentCondition() {
+            return this.isActive && this.component && this.condition && !this.popover;
+        },
+        /** Props to pass to the component of this action. */
+        componentProps: action.componentProps,
+        /** Condition to display this action. */
+        get condition() {
```

Comment:
I would have called it “predicate”

Reply:
You can make a follow up PR to rename it if we think it is better.